code/modules/hydroponics/grown/flowers.dm

```diff
@@ -236,9 +236,10 @@
 	force = round((5 + seed.potency / 5), 1)
 
 /obj/item/grown/novaflower/attack(mob/living/carbon/M, mob/user)
-	if(!..())
-		return
+	. = ..()
 	if(isliving(M))
+		if(HAS_TRAIT(user, TRAIT_PACIFISM))
+			return
```

Comment:
inheritance has already ran and this value will reflect inheritance rather than being a stop in the attack chain (unless inheritance was stopped).

Is that what you want?

Reply:
from what I've tested, without this it will not attack but put people on fire anyway if you are a pacifist, if you have a better fix please tell me since I'm not used to deal with traits.

Reply:
imagine this as `return .`

if inheritance is true, this will return true. I'm asking if that is what you want or are you looking for something like `return FALSE` here.

Reply:
I still don't really get what you mean by "inheritance" is the problem